From ae68be312619ad3bcb2d550a6267d38cce3284f5 Mon Sep 17 00:00:00 2001
From: Adam Jackson <ajax@redhat.com>
Date: Fri, 13 Feb 2009 17:25:32 -0500
Subject: [PATCH] Input: Don't re-block SIGIO in xf86SigioReadInput()

It's already blocked on the way in, because we asked for it to be
blocked at sigaction() time.
---
 hw/xfree86/common/xf86Events.c |    7 ++-----
 1 files changed, 2 insertions(+), 5 deletions(-)

diff --git a/hw/xfree86/common/xf86Events.c b/hw/xfree86/common/xf86Events.c
index 592d4e3..9ed2884 100644
--- a/hw/xfree86/common/xf86Events.c
+++ b/hw/xfree86/common/xf86Events.c
@@ -304,16 +304,13 @@ xf86Wakeup(pointer blockData, int err, pointer pReadmask)
  *    signal handler for the SIGIO signal.
  */
 static void
-xf86SigioReadInput(int fd,
-		   void *closure)
+xf86SigioReadInput(int fd, void *closure)
 {
     int errno_save = errno;
-    int sigstate = xf86BlockSIGIO();
-    InputInfoPtr pInfo = (InputInfoPtr) closure;
+    InputInfoPtr pInfo = closure;
 
     pInfo->read_input(pInfo);
 
-    xf86UnblockSIGIO(sigstate);
     errno = errno_save;
 }
 
-- 
1.7.0.1

